include/libcyphal/transport/msg_sessions.hpp

```diff
@@ -49,14 +47,13 @@ class IMessageTxSession : public IRunnable
     ///
     /// @param metadata Additional metadata associated with the message.
     /// @param payload_fragments Segments of the message payload.
-    /// @return `void` in case of success; otherwise an error.
+    /// @return `nullopt` in case of success; otherwise a transport error.
     ///
-    CETL_NODISCARD virtual Expected<void, AnyError> send(const TransferMetadata& metadata,
+    CETL_NODISCARD virtual cetl::optional<AnyError> send(const TransferMetadata& metadata,
```

Comment:
Making `void` specialization for `Expected` in my near future todo list. Temporary just using `optional`.